src/document/proxy/array_proxy.ts

```diff
@@ -220,18 +220,23 @@ export class ArrayProxy {
           return target.length;
         } else if (typeof method === 'symbol' && method === Symbol.iterator) {
           return ArrayProxy.iteratorInternal.bind(this, context, target);
+        } else if (method === 'find' || method === 'forEach') {
+          return (...args: Array<any>) => {
+            const arr = Array.from(target).map((e) => toProxy(context, e));
+            return arr[method](...args);
+          };
```

Comment:
For now, I also don't know how to handle types reliably for this case, but I've confirmed that it runs without any problems when I try to like the one below.

```suggestion
        } else if (method === 'find' || method === 'forEach' || method === 'filter') {
          return (...args: any) => {
            const arr = Array.from(target).map((elem) => toProxy(context, elem));
            return Array.prototype[method].apply(arr, args);
          };
```

Additionally, when I moved `filter` here, it was fine, but `reduce` caused a type problem.

Reply:
Thank you for the comments. I applied the code you suggested.